src/main/java/dev/slimevr/vr/trackers/IMUTracker.java

```diff
@@ -116,7 +154,12 @@ public boolean getPosition(Vector3f store) {
 	
 	@Override
 	public boolean getRotation(Quaternion store) {
-		store.set(rotQuaternion);
+		if(movementFilterTickCount > 0 && movementFilterAmount != 1 && previousRots.getFirst() != null){
+			store.set(new Quaternion().slerp(rotQuaternion, previousRots.getFirst(), movementFilterAmount));
```

Comment:
Is there any way to reduce instantiating objects? Considering the tick count should be a fixed number, maybe this could be an array of quaternions that are looped over instead?

Reply:
Yeah true imma do that

Reply:
New change looks good 👍